relay-metrics/src/aggregation.rs

```diff
@@ -745,6 +754,31 @@ impl BucketKey {
         std::hash::Hash::hash(self, &mut hasher);
         hasher.finalize() as i64
     }
+
+    /// Remove invalid characters from tags and metric names.
```

Comment:
Metric names and metric tags have different character allow lists, let's have separate validator functions. Especially, there is already [`is_valid_name`](https://github.com/getsentry/relay/blob/e9c9dc9a36b26c450737fb3461c2486605802b99/relay-metrics/src/protocol.rs#L411-L423). [`parse_tags`](https://github.com/getsentry/relay/blob/e9c9dc9a36b26c450737fb3461c2486605802b99/relay-metrics/src/protocol.rs#L485) doesn't have such logic yet.

More of a code style suggestion: `BucketKey` is a plain data object used for bucketing. I would suggest to put such validator functions 
on types that specify the public protocol or directly in the aggregator if it's easily possible.